packages/js/product-editor/src/components/iframe-editor/iframe-editor.tsx

```diff
@@ -42,20 +43,31 @@ type IframeEditorProps = {
 };
 
 export function IframeEditor( {
-	closeModal = () => {},
 	initialBlocks = [],
 	onChange = () => {},
 	onClose,
 	onInput = () => {},
 	settings: __settings,
 }: IframeEditorProps ) {
 	const [ resizeObserver ] = useResizeObserver();
-	const [ blocks, setBlocks ] = useState< BlockInstance[] >( initialBlocks );
 	const [ temporalBlocks, setTemporalBlocks ] =
 		useState< BlockInstance[] >( initialBlocks );
+
+	// Pick the blocks from the store.
+	const blocks: BlockInstance[] = useSelect( ( select ) => {
+		return select( productEditorUiStore ).getModalEditorBlocks();
+	}, [] );
+
+	const {
+		setModalEditorBlocks: setBlocks,
+		closeModalEditor,
+		setModalEditorContentHasChanged,
+	} = useDispatch( productEditorUiStore );
```

Comment:
This feels very specific to the ModalEditor component, would this not be better handled within the `ModalEditor` component? or in the BlockEditor component where the ModalEditor is being used?

The IFrameEditor is kind of a stand alone thing, and should not need to worry/care about the context of its parent ( the modal editor ).